debug.h

```diff
@@ -23,24 +23,44 @@
  */
 void end();
 
+#ifdef ARDUINO
+
 #define ASSERT(exp, ...)                                           \
     {                                                              \
         if (!(exp)) {                                              \
             printf("Assert Failed (%s:%d): ", __FILE__, __LINE__); \
             printf(__VA_ARGS__);                                   \
-            printf("\n\n\nlooping...\n");                          \
-            end();                                                 \
+            exit(-1);                                              \
```

Comment:
if I recall correctly we had an inf loop here to ensure that the message could be read and that the serial output was not just spammed with restarts.